File: frontend/src/types/priorAuthorization.ts
export type PriorAuthorizationStatus =
  | "PENDING_REVIEW"
  | "MORE_INFORMATION_REQUIRED"
  | "ESCALATED"
  | "APPROVED"
  | "DENIED";

export type PriorAuthorizationDecisionAction =
  | "APPROVE"
  | "REQUEST_MORE_INFORMATION"
  | "ESCALATE"
  | "DENY";

export interface PriorAuthorization {
  id: string;
  authorization_number: string;

  member_id: string;
  provider_id: string;
  enrollment_id: string | null;

  diagnosis_code: string | null;
  diagnosis_description: string | null;

  procedure_code: string | null;
  procedure_description: string;

  requested_service_date: string | null;

  priority: string;
  status: PriorAuthorizationStatus;

  clinical_summary: string | null;

  coverage_status: string | null;
  benefit_category: string | null;
  service_covered: string | null;
  authorization_required: string | null;

  ai_recommendation: string | null;
  ai_confidence: number | null;
  medical_necessity_score: number | null;
  ai_rationale: string | null;

  assigned_reviewer: string | null;
  review_due_at: string | null;

  final_decision: string | null;
  decision_rationale: string | null;

  information_requested: string | null;

  escalation_reason: string | null;
  escalated_to: string | null;

  decided_by: string | null;
  decided_at: string | null;

  created_at: string;
  updated_at: string;
}

export interface PriorAuthorizationDecisionRequest {
  action: PriorAuthorizationDecisionAction;
  reviewer: string;
  rationale: string;

  information_requested?: string | null;
  escalation_reason?: string | null;
  escalated_to?: string | null;
}

File: backend/app/schemas/prior_authorization.py
from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class PriorAuthorizationBase(BaseModel):
    member_id: UUID
    provider_id: UUID
    enrollment_id: UUID | None = None

    diagnosis_code: str | None = None
    diagnosis_description: str | None = None

    procedure_code: str | None = None
    procedure_description: str

    requested_service_date: date | None = None

    priority: str = "ROUTINE"

    clinical_summary: str | None = None

    coverage_status: str | None = None
    benefit_category: str | None = None
    service_covered: str | None = None
    authorization_required: str | None = None

    ai_recommendation: str | None = None
    ai_confidence: float | None = Field(
        default=None,
        ge=0,
        le=100,
    )
    medical_necessity_score: float | None = Field(
        default=None,
        ge=0,
        le=100,
    )
    ai_rationale: str | None = None

    assigned_reviewer: str | None = None
    review_due_at: datetime | None = None


class PriorAuthorizationCreate(PriorAuthorizationBase):
    authorization_number: str


class PriorAuthorizationUpdate(BaseModel):
    diagnosis_code: str | None = None
    diagnosis_description: str | None = None

    procedure_code: str | None = None
    procedure_description: str | None = None

    requested_service_date: date | None = None

    priority: str | None = None
    status: str | None = None

    clinical_summary: str | None = None

    coverage_status: str | None = None
    benefit_category: str | None = None
    service_covered: str | None = None
    authorization_required: str | None = None

    ai_recommendation: str | None = None
    ai_confidence: float | None = Field(
        default=None,
        ge=0,
        le=100,
    )
    medical_necessity_score: float | None = Field(
        default=None,
        ge=0,
        le=100,
    )
    ai_rationale: str | None = None

    assigned_reviewer: str | None = None
    review_due_at: datetime | None = None


class PriorAuthorizationDecision(BaseModel):
    action: str
    reviewer: str
    rationale: str | None = None
    information_requested: str | None = None
    escalation_reason: str | None = None
    escalated_to: str | None = None


class PriorAuthorizationResponse(PriorAuthorizationBase):
    model_config = ConfigDict(
        from_attributes=True,
    )

    id: UUID
    authorization_number: str
    status: str

    final_decision: str | None = None
    decision_rationale: str | None = None

    information_requested: str | None = None

    escalation_reason: str | None = None
    escalated_to: str | None = None

    decided_by: str | None = None
    decided_at: datetime | None = None

    created_at: datetime
    updated_at: datetime